
enum e_asc_proto_version{
	ASC_VERSION = 1;
};

//客户端发给as的 有response和notify
enum e_msg_c_as{
	e_msg_c_as_min = 0 ;		//必须为0  从0开始 方便为了funcmap数组存储
	e_msg_c_as_register_req = 1;	//注册请求
	e_msg_c_as_login_req = 2;	//登录请求
	e_msg_c_as_enter_area_req = 3;	//区服信息请求
	e_msg_c_as_logout_req = 4;	//注销请求
	e_msg_c_as_login_if_no_register_req = 5;	//登陆请求 如果不存在则注册
	e_msg_c_as_max = 6;		
};

//as发给客户端的 有response和notify
enum e_msg_as_c{
	e_msg_as_c_min = 0 ;		//必须为0  从0开始 方便为了funcmap数组存储
	e_msg_as_c_register_res = 1 ;	
	e_msg_as_c_login_res = 2 ;
	e_msg_as_c_enter_area_res = 3;
	e_msg_as_c_logout_res = 4;
	e_msg_as_c_login_if_no_register_res = 5;	//登陆请求 如果不存在则注册
	e_msg_as_c_max = 6;
};

enum e_asc_errorcode{
	e_asc_errorcode_min	= 0 ;			//因为初始化就为0 所以这里作为初始化
	e_asc_errorcode_success = 1 ;		//操作成功
	e_asc_errorcode_unkown_error = 2 ;		//未知错误
	e_asc_errorcode_registe_already = 3;		//已经注册 不能重复注册！
	e_asc_errorcode_account_not_exist_or_wrong_password = 4;	//账户不存在或者密码错误
	e_asc_errorcode_not_logined = 5;	//还没有登陆！
	e_asc_errorcode_no_this_area = 6;	//没有所选择的区
	e_asc_errorcode_area_closed = 7;	//所选择的区已经关闭
	e_asc_errorcode_wrong_password = 8;	//登陆失败 账户密码错误！
	e_asc_errorcode_max = 9;
};

//<<< common -------------

message c_as_register_req{
	required int32 plantform_id = 1;	//渠道平台id
	required int32 plantform_sub_id = 2;	//渠道平台的子类型id 为了细分 比如官方 有 pc官网 手机官网
	required string account = 3;	//账户
	required string password = 4;	//密码
};
message as_c_register_res{
	required int32 error_code = 1;	//错误码
	//optional int32 user_id = 2;	//用户id
	//optional bytes	token = 3;	//令牌
};

//账户存在才能登陆成功 否则报错
message c_as_login_req{
	required int32 plantform_id = 1;	//渠道平台id
	required int32 plantform_sub_id = 2;	//渠道平台的子类型id 为了细分 比如官方 有 pc官网 手机官网
	required string account = 3;	//账户
	required string password = 4;	//密码
};

message area{

	enum EAreaStatus{
		EAreaStatus_None = 0 ;		//停服 或者 还未开服
		EAreaStatus_Good = 1 ;		//通畅
		EAreaStatus_Normal = 2 ;	//正常
		EAreaStatus_Bad = 3 ;		//爆满
		EAreaStatus_New = 4 ;		//新区
	};

	required int32 area_id = 1;		//服务器ID
	required string name = 2;	//服务器名字
	optional int32 status = 3;		//服务器性能状态
};

message area_list{
	repeated area areas = 1;//登录成功时 给予区服信息
}

message as_c_login_res{
	required int32 error_code = 1;	//错误码
	//optional int32 user_id = 2;	//用户id
	//optional bytes	token = 3;	//令牌
	optional area_list areas = 3;//区服列表
};


//登陆时 假如账户不存在 就创建
message c_as_login_if_no_register_req{
	required int32 plantform_id = 1;	//渠道平台id
	required int32 plantform_sub_id = 2;	//渠道平台的子类型id 为了细分 比如官方 有 pc官网 手机官网
	required string account = 3;	//账户
	required string password = 4;	//密码
};

//c_as_login_if_no_register_req返回的响应是as_c_login_res 而不是这里的as_c_login_if_no_register_res
//这里的此结构 仅作测试备用
message as_c_login_if_no_register_res{
	required int32 error_code = 1;	//错误码
	//optional int32 user_id = 2;	//用户id
	//optional bytes	token = 3;	//令牌
	optional area_list areas = 2;//区服列表
};

message c_as_enter_area_req{
	required int32 area_id = 1;
};

message as_c_enter_area_res{
	required int32 error_code = 1;	//错误码
	//>>> 成功的时候才会有以下信息
	optional string ip = 2; //	目标区服gs的IP
	optional int32 port = 3;	//目前区服gs的端口
	optional bytes	token = 4;	//令牌
};

message c_as_logout_req{
	required int32 plantform_id = 1;	//渠道平台id
	required int32 plantform_sub_id = 2;	//渠道平台的子类型id 为了细分 比如官方 有 pc官网 手机官网
	required string account = 3;	//账户
	required string password = 4;	//密码
};

message as_c_logout_res{
	optional int32 error_code = 1;	//错误码
};

